### PR TITLE
oneko: 1.2.sakura.5 -> 1.2.hanami.6, and change upstream

### PR DESCRIPTION
**Pull request by IreneKnapp**
###### Motivation for this change
As of approximately a year ago, I am the upstream maintainer of Oneko. The previous version was released in 1999. Per the README, every version has been by a different maintainer. I did attempt to contact the previous one, but since there was no response, I think it's entirely in the spirit of the project to consider my version the latest. I like to think of it as similar to those piles of stones, where each hiker who passes adds one more.

###### Things done
Update Oneko to version 1.2.hanami.6. Add myself as a second nixpkgs maintainer for it.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xaverdh** [comment]
So while I very much appreciate the work, the license part makes this pr somewhat questionable.
I think some people might not want unfree (even if it is redistributable) software on their systems, so I think that we will want to either
* keep both the original and the new one, so ppl can choose
* or you may want to change the license of the new version to something like [gpl,mit,bsd,..] (not sure what your stance on this is)

**Comment by IreneKnapp** [comment]
Yeah, so I added that not because I've changed anything but because I think it was incorrectly classified to begin with. It's one of those strange copyright scenarios that were common in the 90s, before everyone got used to digital rights stuff. My code additions are released into the public domain, but unfortunately the software has always contained various trademarked and copyrighted characters. The README documents that the owners of the art have given permission to use it, but I don't think it's properly described as free.

If I'm being too nitpicky, I'm happy to drop the unfreeRedistributable and let things to back to the status quo of not being examined too closely... I think this program is culturally significant and I don't want to cause trouble for it by asking inconvenient questions.

**Comment by xaverdh** [comment]
Ah, thanks for explaining the reasoning!
debian classifies it as public domain (https://metadata.ftp-master.debian.org/changelogs//main/o/oneko/oneko_1.2.sakura.6-14_copyright) and they are usually *very* careful with these kind of things, so it should be fine I think.

**Comment by IreneKnapp** [comment]
I appreciate your diligence. I put this together a while ago and had forgotten I added that part. Sorry about the confusion. Anyway, I set it back to just publicDomain; please take a look. :)

**Review by xaverdh — APPROVED** [review]
looks good!
You probably want to squash some of the commits before this will be merged.

**Review by xaverdh — APPROVED** [review]

File: pkgs/applications/misc/oneko/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/oneko/default.nix:10]:
```suggestion
    rev = version_name;
```

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4326]: Please put this into a separate commit with the message ``maintainers: add irenes`` and sort it alphabetically.

Comment (IreneKnapp) [reply maintainers/maintainer-list.nix:4326]: done; please take a look.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**
